src/extension/importer/core/src/main/java/org/geoserver/importer/FileData.java

```diff
@@ -28,7 +29,7 @@
     protected Resource file;
 
     public FileData(Resource file) {
-        this.file = file;
+        this.file = Resources.serializable(file);
```

Comment:
It looks like this got missed when the ResourceStore work was initially done, but FileData is a representation of a file that is being loaded into Importer (in other words, not yet added to geoserver), and should actually be using File, not Resource.